syntax = "proto3";

message FromEmbedded {
  oneof contents {
    CanDataChunk canDataChunk = 1;
    bytes gpsDataChunk = 2;
    LogFSInfo logFSInfo = 3;
  }
}

message ToEmbedded {
  ControlCommand command = 1;
  CanDataChunk transmitData = 2;
}

message ControlCommand { ControlCommandType commandType = 1; }

message LogFSInfo { repeated LogFileInfo logFiles = 1; }

message CanDataChunk {
  uint32 id = 1;
  bytes data = 2;
}

message LogFileInfo {
  uint32 size = 1;
  uint64 unixTime = 2;
  string fileName = 3;
}

enum ControlCommandType {
  NO_CMD = 0;
  STOP_LOG = 1;
  START_LOG = 2;
  MARK_LOG = 3;
  GET_LOGFS_INFO = 4;
  GET_LOG = 5;
  DELETE_LOG = 6;
}